pkg/orchestrator/scanestimationwatcher/watcher.go

```diff
@@ -0,0 +1,566 @@
+	logger := log.GetLoggerFromContextOrDiscard(ctx)
+
+	if scanEstimation == nil {
+		return errors.New("invalid Scan estimation: object is nil")
+	}
+
+	scanEstimationID, ok := scanEstimation.GetID()
+	if !ok {
+		return errors.New("invalid Scan estimation: Id is nil")
+	}
+
+	if err := w.createAssetScanEstimationsForScanEstimation(ctx, scanEstimation); err != nil {
+		return fmt.Errorf("failed to creates AssetScanEstimation(s) for ScanEstimation. ScanEstimationID=%s: %w", scanEstimationID, err)
+	}
+	scanEstimation.State = utils.PointerTo(models.ScanEstimationStateInProgress)
+
+	scanEstimationPatch := &models.ScanEstimation{
+		State:    scanEstimation.State,
+		Summary:  scanEstimation.Summary,
+		AssetIDs: scanEstimation.AssetIDs,
+	}
+	err := w.backend.PatchScanEstimation(ctx, scanEstimationID, scanEstimationPatch)
+	if err != nil {
+		return fmt.Errorf("failed to update Scan estimation. ScanEstimationID=%s: %w", scanEstimationID, err)
+	}
+
+	logger.Infof("Total %d unique assets for ScanEstimation", len(*scanEstimation.AssetIDs))
+
+	return nil
+}
+
+func (w *Watcher) createAssetScanEstimationsForScanEstimation(ctx context.Context, scanEstimation *models.ScanEstimation) error {
+	logger := log.GetLoggerFromContextOrDiscard(ctx)
+
+	if scanEstimation.AssetIDs == nil || *scanEstimation.AssetIDs == nil {
+		return nil
+	}
+	numOfAssets := len(*scanEstimation.AssetIDs)
+
+	errs := make(chan error, numOfAssets)
+	var wg sync.WaitGroup
+	for _, id := range *scanEstimation.AssetIDs {
+		wg.Add(1)
+		assetID := id
+		go func() {
+			defer wg.Done()
+
+			err := w.createAssetScanEstimationForAsset(ctx, scanEstimation, assetID)
+			if err != nil {
+				logger.WithField("AssetID", assetID).Errorf("Failed to create AssetScanEstimation: %v", err)
+				errs <- err
+
+				return
+			}
+		}()
+	}
+	wg.Wait()
+	close(errs)
+
+	assetErrs := make([]error, 0, numOfAssets)
+	for err := range errs {
+		assetErrs = append(assetErrs, err)
+	}
+	numOfErrs := len(assetErrs)
+
+	if numOfErrs > 0 {
+		return fmt.Errorf("failed to create %d AssetScanEstimation(s) for ScanEstimation. ScanEstimationID=%s: %w", numOfErrs, *scanEstimation.Id, assetErrs[0])
+	}
+
+	if scanEstimation.Summary == nil {
+		scanEstimation.Summary = &models.ScanEstimationSummary{}
+	}
+	scanEstimation.Summary.JobsLeftToRun = utils.PointerTo(numOfAssets)
+
+	return nil
+}
+
+func (w *Watcher) newAssetScanEstimationFromScanEstimation(ctx context.Context, scanEstimation *models.ScanEstimation, assetID string) (*models.AssetScanEstimation, error) {
+
+	if scanEstimation.ScanTemplate == nil {
+		return nil, fmt.Errorf("empty scan template")
+	}
+
+	// get asset from backend
+	asset, err := w.backend.GetAsset(ctx, assetID, models.GetAssetsAssetIDParams{})
+	if err != nil {
+		return nil, err
+	}
+
+	assetWriteable := &models.Asset{
+		Id:         asset.Id,
+		Revision:   asset.Revision,
+		ScansCount: asset.ScansCount,
+		Summary:    asset.Summary,
+		AssetInfo:  asset.AssetInfo,
+	}
```

Comment:
This should be a relationship to an asset in the system, we should not be duplicating any asset information in the AssetScanEstimation.